nomos-mantle/src/tx.rs

```diff
@@ -68,9 +68,30 @@ impl nomos_core::tx::Transaction for MantleTx {
     }
 }
 
-#[derive(Serialize, Deserialize)]
+impl From<SignedMantleTx> for MantleTx {
+    fn from(signed_tx: SignedMantleTx) -> Self {
+        signed_tx.mantle_tx
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
 pub struct SignedMantleTx {
     pub mantle_tx: MantleTx,
     pub ops_profs: Vec<OpProof>,
     pub ledger_tx_proof: ZkSignature,
 }
+
+impl Hash for SignedMantleTx {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.mantle_tx.hash(state);
+    }
+}
+
+impl nomos_core::tx::Transaction for SignedMantleTx {
+    const HASHER: TransactionHasher<Self> = |tx| blake2::Blake2b::digest(tx.as_bytes()).into();
+    type Hash = TxHash;
+
+    fn as_bytes(&self) -> bytes::Bytes {
+        self.mantle_tx.as_bytes()
```

Comment:
why just the mantle tx?

Reply:
But then it should not be implemented for the Signed version of it (?)

Reply:
`Transaction` trait is what consensus and mempool are expecting. If we pass SignedMantleTx to consensus and mempool, then `SignedMantleTx` should implement `Transaction`. Transaction trait implementation is actually not needed for the MantleTx, at least it's not used anywhere.
In the end I see both of these transactions as one - they are related via one thing - hash.

Reply:
> are Hash and as_bytes tied together

Yes, the `Transaction` trait is defined in [`nomos-core`](https://github.com/logos-co/nomos/blob/master/nomos-core/chain-defs/src/tx/mod.rs#L14-L23). Maybe you had different idea for `Transaction` trait when initially defining, but at the moment there's no other use for it than to get the hash.

I agree that `as_bytes` is bit confusing, there were some ideas to rename this trait to `TransactionHash` or similar, but then `as_bytes` method wouldn't make sense even more. I suggest to keep it as is - if this trait is implemented - we can get 1) hash 2) transaction bytes.

Reply:
I think currently it's quite confusing as `as_bytes` is a common method name in Rust that carries some semantics. What about `as_sign_bytes` implying those are the bytes to be signed?